packages/node-xmpp-server/lib/S2S/Router.js

```diff
@@ -34,9 +34,14 @@ function Router (s2sPort, bindAddress, opts) {
   }
 
   // start tcp socket
-  net.createServer(function (socket) {
+  var server = net.createServer(function (socket) {
     this.acceptConnection(socket)
   }.bind(this)).listen(s2sPort || 5269, bindAddress || '::')
+
+  this.close = function (callback) {
+    debug('closing server')
```

Comment:
For c2s we have close and shutdown/end

close closes the port and shutdown/end closes the port and end all connections

Checkout serverStop.js (there's a bug but the idea is similar) maybe you can use it for Router as well.

On June 2, 2016 8:55:10 PM GMT+02:00, Xiaoxin Lu <notifications@github.com> wrote:
>>      this.acceptConnection(socket)
>>    }.bind(this)).listen(s2sPort || 5269, bindAddress || '::')
>> +
>> +  this.close = function (callback) {
>> +    debug('closing server')
>
>it closes the server off to new incoming connections. It doesn't affect
>the outgoing connection nor existing [incoming
>connections](https://nodejs.org/api/net.html#net_server_close_callback).
>Maybe `closeServer` would be a better name?
>
>---
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/node-xmpp/node-xmpp/pull/304/files/e6daf61bef9011b188acb374b601fd02c498f0da#r65597731

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.